src/test/kotlin/no/nav/medlemskap/clients/aareg/AaregMapperTest.kt

```diff
@@ -0,0 +1,596 @@
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.medlemskap.clients.ereg.Ansatte
+import no.nav.medlemskap.clients.ereg.Bruksperiode
+import no.nav.medlemskap.domene.Arbeidsforholdstype
+import no.nav.medlemskap.domene.Periode
+import no.nav.medlemskap.services.aareg.AaRegService
+import no.nav.medlemskap.services.aareg.mapAaregResultat
+import org.junit.Assert
+import org.junit.Test
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+class AaregMapperTest {
+
+    @Test
+    fun mapAaregResultatTilArbeidsforhold() {
+
+        val javaTimeModule = JavaTimeModule()
+        val localDateTimeDeserializer = LocalDateTimeDeserializer(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS"))
+        javaTimeModule.addDeserializer(LocalDateTime::class.java, localDateTimeDeserializer)
+
+        val objectMapper: ObjectMapper = ObjectMapper()
+            .registerModules(KotlinModule(), javaTimeModule)
+            .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+
+        val aaRegArbeidsforhold = objectMapper.readValue<AaRegArbeidsforhold>(jsonStringAaRegArbeidsforholdList)
+
+        val dataOmArbeidsgiver = mutableMapOf<String, AaRegService.ArbeidsgiverInfo>()
+        val juridiskEnhetOrgnummerEnhetstype = mutableMapOf<String, String?>()
+        juridiskEnhetOrgnummerEnhetstype["123456789"] = "AS"
+        val orgnummer = "985672744"
+        dataOmArbeidsgiver[orgnummer] = AaRegService.ArbeidsgiverInfo(
+            arbeidsgiverEnhetstype = "BEDR",
+            ansatte = listOf(Ansatte(5, Bruksperiode(LocalDate.now(), LocalDate.now().plusDays(2)), null)),
+            opphoersdato = null,
+            konkursStatus = null,
+            juridiskEnhetOrgnummerEnhetstype = juridiskEnhetOrgnummerEnhetstype
+        )
+
+        val mappedAaregResultatList = mapAaregResultat(listOf(aaRegArbeidsforhold), dataOmArbeidsgiver)
+        Assert.assertEquals(1, mappedAaregResultatList.size)
+
+        val mappedAaregResultat = mappedAaregResultatList.first()
+        Assert.assertEquals("985672744", mappedAaregResultat.arbeidsgiver.identifikator)
+        Assert.assertEquals("BEDR", mappedAaregResultat.arbeidsgiver.type)
+        Assert.assertEquals(5, mappedAaregResultat.arbeidsgiver.ansatte!!.first().antall)
+        Assert.assertEquals("AS", mappedAaregResultat.arbeidsgiver.juridiskEnhetEnhetstypeMap?.get("123456789"))
+        Assert.assertEquals(1, mappedAaregResultat.arbeidsavtaler.size)
+        Assert.assertEquals("Organisasjon", mappedAaregResultat.arbeidsgivertype.name)
+        Assert.assertEquals(100.0, mappedAaregResultat.arbeidsavtaler.first().stillingsprosent)
+        Assert.assertEquals(Arbeidsforholdstype.NORMALT, mappedAaregResultat.arbeidsfolholdstype)
+        Assert.assertEquals(Periode(LocalDate.parse("2008-01-01"), LocalDate.parse("2018-10-30")), mappedAaregResultat.periode)
+        Assert.assertNull(mappedAaregResultat.utenlandsopphold)
```

Comment:
Samme som over.

Hadde det kanskje vært greit å ha en test som ser på om utenlandsopphold blir mappet dersom det finnes?

Reply:
Utenlandsopphold brukes ikke i koden, men hvis jeg ikke husker feil så kan det hende det vil være bruk for den informasjonen på et senere tidspunkt. Jeg legger det i testen.